False positive of extension_access_modifier on property with private(set)

### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

When a `public` extension has a single property with `private(set)`, the property receives a warning that it should use an extension access modifier.

This isn't possible since private(set) cannot be moved from the property to the extension.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
Linting Swift files in current working directory
Linting 'test.swift' (1/1)
test.swift:3:5: warning: Extension Access Modifier Violation: Prefer to use extension access modifiers (extension_access_modifier)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.55.1
* Installation method used (Homebrew, CocoaPods, building from source, etc)? homebrew
* Paste your configuration file: n/a

```yml
opt_in_rules:
  - extension_access_modifier
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
  If so, paste their relative paths and respective contents.
* Which Xcode version are you using (check `xcodebuild -version`)? 15.4
* Do you have a sample that shows the issue? Yes

```swift
public class Foo {}
public extension Foo {
    private(set) var value: Int {
        get { 1 }
        set {}
    }
}
```

#### Single line to run it

```shell
echo "class Foo {}\npublic extension Foo {\n    private(set) var value {\n        get { 0 }\n        set { print(newValue) }\n    }\n}" | swiftlint lint --no-cache --use-stdin --enable-all-rules  
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -26,6 +26,11 @@
   [SimplyDanny](https://github.com/SimplyDanny)
   [#5153](https://github.com/realm/SwiftLint/issues/5153)
 
+* Ignore access level modifiers restricted to value setting in
+  `extension_access_modifier` rule.  
+  [SimplyDanny](https://github.com/SimplyDanny)
+  [#5623](https://github.com/realm/SwiftLint/issues/5623)
+
 ## 0.55.1: Universal Washing Powder
 
 #### Breaking

--- a/Source/SwiftLintBuiltInRules/Rules/Idiomatic/ExtensionAccessModifierRule.swift
+++ b/Source/SwiftLintBuiltInRules/Rules/Idiomatic/ExtensionAccessModifierRule.swift
@@ -97,6 +97,11 @@ struct ExtensionAccessModifierRule: OptInRule {
                 set { Foo.shared.bar = newValue }
               }
             }
+            """),
+            Example("""
+            public extension Foo {
+              private(set) var value: Int { 1 }
+            }
             """)
         ],
         triggeringExamples: [
@@ -156,6 +161,11 @@ struct ExtensionAccessModifierRule: OptInRule {
               ↓private func bar() {}
               ↓private func baz() {}
             }
+            """),
+            Example("""
+            ↓extension Foo {
+              private(set) public var value: Int { 1 }
+            }
             """)
         ]
     )
@@ -198,7 +208,7 @@ private extension ExtensionAccessModifierRule {
 
             for decl in node.memberBlock.expandingIfConfigs() {
                 let modifiers = decl.asProtocol((any WithModifiersSyntax).self)?.modifiers
-                let aclToken = modifiers?.accessLevelModifier?.name
+                let aclToken = modifiers?.accessLevelModifier()?.name
                 let acl = ACL.from(tokenKind: aclToken?.tokenKind)
                 if areAllACLsEqual, acl != aclTokens.last?.acl, aclTokens.isNotEmpty {
                     areAllACLsEqual = false

--- a/Source/SwiftLintCore/Extensions/SwiftSyntax+SwiftLint.swift
+++ b/Source/SwiftLintCore/Extensions/SwiftSyntax+SwiftLint.swift
@@ -100,6 +100,16 @@ public extension DeclModifierListSyntax {
         first { $0.asAccessLevelModifier != nil }
     }
 
+    func accessLevelModifier(setter: Bool = false) -> DeclModifierSyntax? {
+        first {
+            if $0.asAccessLevelModifier == nil {
+                return false
+            }
+            let hasSetDetail = $0.detail?.detail.tokenKind == .identifier("set")
+            return setter ? hasSetDetail : !hasSetDetail
+        }
+    }
+
     func contains(keyword: Keyword) -> Bool {
         contains { $0.name.tokenKind == .keyword(keyword) }
     }

